tools/clang/lib/SPIRV/DeclResultIdMapper.cpp

```diff
@@ -2041,6 +2041,10 @@ bool DeclResultIdMapper::createStageVars(
                                                  clang::ArrayType::Normal, 0);
       break;
     case hlsl::Semantic::Kind::InnerCoverage:
+      if (!type->isSpecificBuiltinType(BuiltinType::UInt)) {
```

Comment:
I am curious why this is not reported by the front-end parser. Anyway, it looks good to me.

Reply:
You're right. Unfortunately this is also an issue when compiling to DXIL. So it's in fact better if the frontend performs the check. I'll merge this PR, and I'll check if I can modify the frontend in a separate PR. Thanks